cmd/neofs-cli/internal/common/exit.go

```diff
@@ -3,25 +3,32 @@ package common
 import (
 	"errors"
 	"fmt"
-	"os"
 
+	"github.com/nspcc-dev/neofs-node/cmd/internal/custom_errors"
 	sdkstatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
-	"github.com/spf13/cobra"
 )
 
 // ErrAwaitTimeout represents the expiration of a polling interval
 // while awaiting a certain condition.
 var ErrAwaitTimeout = errors.New("await timeout expired")
 
-// ExitOnErr prints error and exits with a code depending on the error type
+// WrapError wrap error to custom_errors.ExitErr, if it not already, and add a code depending on the error type
 //
 //	0 if nil
 //	1 if [sdkstatus.ErrServerInternal] or untyped
 //	2 if [sdkstatus.ErrObjectAccessDenied]
 //	3 if [ErrAwaitTimeout]
-func ExitOnErr(cmd *cobra.Command, errFmt string, err error) {
+func WrapError(errFmt string, err error) *custom_errors.ExitErr {
```

Comment:
I'd probably do it as `WrapError(err error)` and leave `fmt.Errorf()` to the caller if and when needed. You still can unwrap internally to get the code, but formatting would be easier, I think. You can have cases like `"smth (additional data: %d): %w"` and this won't fit the current interface.